src/components/node-list/styles/_filter.scss

```diff
@@ -14,72 +39,89 @@
     fill: none;
     stroke: var(--color-nodelist-filter-indicator-off);
   }
-}
 
-// Fade row text when unchecked
-.pipeline-nodelist__row--unchecked .pipeline-nodelist__row__text--kind-filter {
-  opacity: 0.55;
+  .pipeline-nodelist__row:hover &.pipeline-row__toggle-icon--all-unset,
+  [data-whatintent='keyboard'] .pipeline-nodelist__row__checkbox:focus + &,
+  &.pipeline-row__toggle-icon--parent {
+    fill: $color-primary;
+    stroke: $color-primary;
+  }
 }
 
-// Brighter row text when unchecked and hovered
-.pipeline-nodelist__row--unchecked:hover
-  .pipeline-nodelist__row__text--kind-filter {
-  opacity: 0.8;
-}
+//--- Toggle icon opacities ---//
 
-// Bright row text when all unset
-.pipeline-nodelist__group--all-unset
-  .pipeline-nodelist__row--unchecked
-  .pipeline-nodelist__row__text--kind-filter {
-  opacity: 1;
-}
+/*
+ Parent (toggle all tags):
+ | row-hover | icon-hover | checked | opacity | ID |
+ |    🚫     |     🚫      |    🚫   |    0    | 0  |
+ |    ✅     |     🚫      |    🚫   |   0.3   | 1  |
+ |    ✅     |     ✅      |    🚫   |   0.55  | 2  |
+ |    🚫     |     🚫      |    ✅   |   0.55  | 2  |
+ |    ✅     |     ✅      |    ✅   |   0.75  | 3  |
 
-// Hide all indicators when all unset
-.pipeline-nodelist__group--all-unset .pipeline-row__toggle--kind-filter {
-  opacity: 0;
-}
+ Child (individual tag):
+ | row-hover | checked | opacity | ID |
+ |    🚫     |    🚫    |    0    | 0  |
+ |    ✅     |    🚫    |   0.3   | 1  |
+ |    🚫     |    ✅    |   0.55  | 2  |
+ |    ✅     |    ✅    |   0.75  | 3  |
+ */
 
-// Show indicator when all unset and row hovered
-.pipeline-nodelist__group--all-unset
-  .pipeline-nodelist__row:hover
-  .pipeline-row__toggle--kind-filter {
-  opacity: 0.55;
+$icon-opacity-0: 0;
+$icon-opacity-1: 0.3;
+$icon-opacity-2: 0.55;
+$icon-opacity-3: 0.75;
```

Comment:
Other than just defining these opacity variables just for the icons, given that those values are also used for the text labels,  it might be better to specify it just as an opacity variable (and later on applied to the classes that defines the cases for the toggle button and the label).

For example:

`$opacity-0: 0;`
`$opacity-1: 0.3;`
...
`$opacity-3: 0.75;`

Reply:
The text labels use the same 0.55 value, but the rest is completely different (as it uses 0.55 / 0.8 / 1). So I figured since those are a different scale, it didn't make sense for these variables to be used there, and it would be confusing to use just one of them out of context.

I only used this variable schema here for the filter icon opacities because the classes and opacities in this specific case are complicated and I think this schema makes things easier to understand - we don't need to make every single CSS value a variable everywhere throughout the codebase, as abstractions introduce their own overhead. I just felt it was helpful here.